app/controllers/events_controller.rb

```diff
@@ -0,0 +1,75 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: [:show, :edit, :update, :destroy]
+  before_action :authorize_project_member
+  before_action :authorize_event_author, only: [:update, :destroy]
+
+  def index
+    @project = Project.find(params[:project_id])
```

Comment:
why 2 vars? You fetch events table which will have an access to corresponding projects if necessary. Maybe describe from the beggining what you want to achieve ..For me it is very strange..in index action in events controller you fetch only SINGLE project.
The convention is that index represents the collection of objects of a given type.